scripts/dts/edtlib.py

```diff
@@ -1347,6 +1347,10 @@ class Property:
     enum_index:
       The index of the property's value in the 'enum:' list in the binding, or
       None if the binding has no 'enum:'
+
+    enum_tokenizable:
```

Comment:
Is this C tokenizable?

Reply:
IMO it should be a preprocessor token.

Reply:
> Is this C tokenizable? Just wondering if we should be more explicit about what tokenizable means.

It's kind of up for debate, really, since we can always convert to a token by swapping things out for underscores.

I think we should go for the following rule:

1. If all the `enum:` values, after converting non-alphanumerics to underscores, are unique, then we emit `ENUM_TOKEN` defines for them with that conversion done.
2. If all the `enum:` values, after **uppercasing and** converting non-alphanumerics to underscores, are unique, then we emit `ENUM_TOKEN_UPPER` defines for them with that conversion done

So these are 'tokenizable' by rule 1, and the `ENUM_TOKEN` defines would be respectively:

- `foo`, `b a r`, `123-hello` -> `foo`, `b_a_r`, `123_hello`
- `FOO`, `Foo` -> `FOO`, `Foo`

Of those:

- the `foo`, `b a r`, `123-hello` set also gets `ENUM_TOKEN_UPPER` defines `FOO`, `B_A_R`, `123_HELLO` respectively
- `FOO`, `Foo` would not be unique after uppercasing, so no `ENUM_TOKEN_UPPER` macros get defined

These are **not** tokenizable by these rules:

- `foo bar`, `foo-bar`: not unique after converting to `_`

Does that work for people? It would solve my ADC problem and doesn't seem to affect the pm-states PR since it's a generalization of what they're already using. It also lets us map lowercase DT strings to uppercase enumerators in C if we want to.

I will update if so.

Reply:
I can't see any obvious problems with this.  Could a non-error diagnostic be emitted if an enum cannot be tokenized (either style) without creating duplicates?  Because I'm guessing that'd be more likely to be a mistake than intent.

Reply:
>  Could a non-error diagnostic be emitted if an enum cannot be tokenized (either style) without creating duplicates?

Sure, I can add a warning in that case. @galak -- all that seem reasonable?

Reply:
>  @galak -- all that seem reasonable?

I went ahead and implemented this